crates/shared/src/trade_finding/external.rs

```diff
@@ -69,8 +79,10 @@ impl ExternalTradeFinder {
             request = request.header("X-REQUEST-ID", id);
         }
 
-        let future = async {
+        let timeout = self.timeout.clone();
```

Comment:
The deadline on the request should also use this timeout and not `super::time_limit` (otherwise you may have an inconsistency)